auth/api/v1/client.go

```diff
@@ -0,0 +1,64 @@
+package v1
+
+import (
+	"context"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/nuts-foundation/nuts-node/auth"
+	"github.com/nuts-foundation/nuts-node/core"
+)
+
+// HTTPClient holds the server address and other basic settings for the http client
+type HTTPClient struct {
+	ServerAddress string
+	Timeout       time.Duration
+}
+
+func (h HTTPClient) client() ClientInterface {
+	response, err := NewClientWithResponses(h.ServerAddress)
```

Comment:
oauth endpoint registrations also include the path.
Is `NewCLientWithResponses` capable of using nil as server address and then use a complete URL in the request?

Reply:
Yeah, looks like it works. Note that in `HTTPClient.CreateAccessToken` the URL is overriden using a request editor function.